src/generate_reduce.jl

```diff
@@ -0,0 +1,21 @@
+"""
+    generate_min_set(g, gen_func, Reps)
+Generate a vector `Reps` times using `gen_func(g)` and return the vector with the least elements.
+"""
+generate_min_set(g::AbstractGraph{T}, gen_func, Reps::Integer) where T<: Integer =
+mapreduce(gen_func, (x, y)->length(x)<length(y) ? x : y, Iterators.repeated(g, Reps))
```

Comment:
Is Iterators.repeated making `Reps` copies of `g` or is it a reference?

Reply:
It's a reference.
```
julia> g = PathGraph(100)
{100, 99} undirected simple Int64 graph

julia> for gh in Iterators.repeated(g, 100)
       add_vertex!(gh)
       end

julia> g
{200, 99} undirected simple Int64 graph

```

Reply:
I don't understand the purpose of this function.

Reply:
`minimal_dominating_set(g)` will return a random dominating set that satisfies the minimal property.

This function will be used to obtain `Reps` random dominating sets (In parallel) and return the smallest one.